### PR TITLE
Initial Commit of adding new UI designed with vue

### PR DESCRIPTION
**Pull request by ahmedifhaam**
-Only works for local paths
-Only intergrated proteus UI Viz features have to intergrated

### CODE REVIEW

**Comment by chrismattmann** [comment]
this includes generated JS and CSS minified. We need the original source, not the minified stuff. I can't review this.

**Comment by ahmedifhaam** [comment]
@chrismattmann i have added the source

**Comment by chrismattmann** [comment]
whoah - i think you checked in like a fully built version of the project. Can you just check in the source code @ahmedifhaam ? It shouldn't be 2M lines of code....I think that's the built version of the project.

**Comment by ahmedifhaam** [comment]
@chrismattmann Updated Please check again Sorry a bunch

**Comment by chrismattmann** [comment]
will check today @ahmedifhaam

File: webapps/proteus-new/pom.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (chrismattmann) [inline webapps/proteus-new/pom.xml:1]: add ALv2 in comments.

File: webapps/proteus-new/src/main/webapp/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Manifest-Version: 1.0
```
Comment (chrismattmann) [inline webapps/proteus-new/src/main/webapp/META-INF/MANIFEST.MF]: exclude this file.

File: webapps/proteus-new/src/main/webapp/css/app.a2690703.css [hunk 3]
```diff
@@ -0,0 +1 @@
...-right:10%}.projectstable{padding-top:10px;padding-bottom:10px}tr:nth-child(2n){background-color:#f2f2f2}tr{background-color:#ddd}th{background-color:#2196f3}#tablecard{margin-top:20px;padding:10px}#barchart[data-v-18eb8837]{margin-top:5%;margin-bottom:5%}svg{margin:25px}path{fill:none;stroke:#76bf8a;stroke-width:3px}area{fill:#76bf8a}line{stroke:#4f7f5c;stroke-width:1px;fill:none}selector{stroke:#28402e;stroke-width:3px;fill:none}.bar{fill:#4682b4}.bar:hover{fill:brown}#barchart{margin-top:5%;margin-bottom:5%;padding-top:5%;margin-left:5%}.arc text{font:10px sans-serif;text-anchor:middle}.arc path{stroke:transparent}#piecard{margin-top:5%;margin-bottom:5%;padding-top:5%;margin-left:5%}#filelistcard{margin-top:10%}#tile{height:35px}#statisticscard[data-v-0fe817e5]{margin-top:10%;margin-left:10%}#header[data-v-0fe817e5]{background-color:#d3d3d3}#progresscard[data-v-1ef03eaf]{margin-left:10%;margin-top:10%}#app{font-family:Avenir,Helvetica,Arial,sans-serif;-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale;text-align:center;color:#2c3e50;margin-top:0}#contentpane{padding-left:10%;padding-right:10%}#footercard{background-color:#2c3e50;width:100%}#logospace{width:80%}
```
Comment (chrismattmann) [inline webapps/proteus-new/src/main/webapp/css/app.a2690703.css]: this is minified. include the actual source.

File: webapps/proteus-new/src/main/webapp/index.html [hunk 4]
```diff
@@ -0,0 +1 @@
+<!DOCTYPE html><html><head><meta charset=utf-8><meta http-equiv=X-UA-Compatible content="IE=edge"><meta name=viewport content="width=device-width,initial-scale=1"><link rel=stylesheet href="http://fonts.googleapis.com/css?family=Roboto:300,400,500,700,400italic"><link rel=stylesheet href="http://fonts.googleapis.com/icon?family=Material+Icons"><link rel=icon href=favicon.ico><base href=./ ><title>proteus</title><link as=style href=./css/app.a2690703.css rel=preload><link as=style href=./css/chunk-vendors.a1864bcf.css rel=preload><link as=script href=./js/app.fcfd2d43.js rel=preload><link as=script href=./js/chunk-vendors.2221aa85.js rel=preload><link href=./css/chunk-vendors.a1864bcf.css rel=stylesheet><link href=./css/app.a2690703.css rel=stylesheet></head><body><noscript><strong>We're sorry but proteus doesn't work properly without JavaScript enabled. Please enable it to continue.</strong></noscript><div id=app></div><script src=./js/chunk-vendors.2221aa85.js></script><script src=./js/app.fcfd2d43.js></script></body></html>
```
Comment (chrismattmann) [inline webapps/proteus-new/src/main/webapp/index.html]: this is minified. Include the actual document.

File: webapps/proteus-new/src/main/webapp/js/app.fcfd2d43.js [hunk 5]
```diff
@@ -0,0 +1,2 @@
...sscomp:kt},data:function(){return{}},methods:{setHost:function(){this.$log.info(location.origin),y.commit("setOrigin",location.origin)}},computed:{progress:function(){return y.state.progress}},mounted:function(){this.setHost()}},St=Et,At=(n("ZL7j"),Object(h["a"])(St,s,o,!1,null,null,null)),Tt=At.exports,$t=n("hf9i"),It=n.n($t),zt=n("A6WO"),Lt=n.n(zt);n("v0CA"),n("0eeA");r["a"].config.productionTip=!0,r["a"].use(i.a),r["a"].use(Lt.a);var Ft={logLevel:"debug",stringifyArguments:!1,showLogLevel:!1};r["a"].use(It.a,Ft),new r["a"]({store:y,render:function(t){return t(Tt)}}).$mount("#app")},YEKE:function(t,e,n){},ZL7j:function(t,e,n){"use strict";var r=n("EDI0"),a=n.n(r);a.a},"e+R+":function(t,e,n){"use strict";var r=n("5Cyc"),a=n.n(r);a.a},eDQf:function(t,e,n){"use strict";var r=n("qjql"),a=n.n(r);a.a},egf1:function(t,e,n){"use strict";var r=n("iS7a"),a=n.n(r);a.a},iS7a:function(t,e,n){},jGAi:function(t,e,n){"use strict";var r=n("+LEs"),a=n.n(r);a.a},kH1g:function(t,e,n){},n3Gq:function(t,e,n){"use strict";var r=n("2nTd"),a=n.n(r);a.a},owsx:function(t,e,n){"use strict";var r=n("AjMG"),a=n.n(r);a.a},qjql:function(t,e,n){},s0m0:function(t,e,n){"use strict";var r=n("YEKE"),a=n.n(r);a.a}});
```
Comment (chrismattmann) [inline webapps/proteus-new/src/main/webapp/js/app.fcfd2d43.js]: no minified code - if it's vendor code, put it into a vendor folder.

File: webapps/proteus-new/src/main/webapp/resources/src/components/HelloWorld.vue [hunk 6]
```diff
@@ -0,0 +1,38 @@
+<template>
```
Comment (chrismattmann) [inline webapps/proteus-new/src/main/webapp/resources/src/components/HelloWorld.vue]: delete this? this file isn't needed.

File: webapps/proteus-new/src/main/webapp/resources/src/components/barchartcomp.vue [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+    </v-card>
+  </section>
+
+</template>
+
+<script lang="js">
+import * as d3 from 'd3';
+import axios from 'axios';
+import store from './../store/store'
+// const props={
```
Comment (chrismattmann) [inline webapps/proteus-new/src/main/webapp/resources/src/components/barchartcomp.vue]: commented code. Remove.

File: webapps/proteus-new/src/main/webapp/resources/src/components/barchartcomp.vue [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+        setInterval(function () {
+          this.loadData();
+        }.bind(this), 1000);
+        
+    },
+    data() {
+      return {
+        licenseTypes : [], 
+        emptynote : '',
+        // licenseTypes : [{"type":"Binaries","numberOfObjects":139,"weight":0.40406976744186046},{"type":"Apache Licensed","numberOfObjects":96,"weight":0.27906976744186046},{"type":"Generated Documents","numberOfObjects":0,"weight":0.0},{"type":"Archives","numberOfObjects":0,"weight":0.0},{"type":"Notes","numberOfObjects":1,"weight":0.0029069767441860465},{"type":"Standards","numberOfObjects":108,"weight":0.313953488372093}]
```
Comment (chrismattmann) [inline webapps/proteus-new/src/main/webapp/resources/src/components/barchartcomp.vue]: commented code. Remove.

File: webapps/proteus-new/src/main/webapp/resources/src/components/barchartcomp.vue [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+        // dts : [{letter:'A',frequency:0.9167},{letter:'B',frequency:0.4503},{letter:'C',frequency:0.4503}],
+      }
+    },
+    watch:{
+     
+    },
+    methods: {
+        loadData(){
+          // var config = {
+          //   headers: {'Access-Control-Allow-Origin': '*'}
```
Comment (chrismattmann) [inline webapps/proteus-new/src/main/webapp/resources/src/components/barchartcomp.vue]: commented code. Remove.

File: webapps/proteus-new/src/main/webapp/resources/src/components/barchartcomp.vue [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+            .then(response=>{
+              this.$log.info(response.data);
+              this.licenseTypes=response.data;
+              this.init();
+            })
+            .catch(error=>{
+              this.emptynote = error.toString();
+              throw error;
+            })
+          // this.init();
```
Comment (chrismattmann) [inline webapps/proteus-new/src/main/webapp/resources/src/components/barchartcomp.vue]: commented code. Remove.

File: webapps/proteus-new/src/main/webapp/resources/src/components/barchartcomp.vue [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+
+          var g = svg.append("g")
+              .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
+
+          
+          
+          var dataval = this.dts;
+          if(dataval.length!=0){
+            this.emptynote='';
+            //x.domain(d3.extent(dataval,function(d) { return d.letter; }));
```
Comment (chrismattmann) [inline webapps/proteus-new/src/main/webapp/resources/src/components/barchartcomp.vue]: commented code. Remove.

File: webapps/proteus-new/src/main/webapp/resources/src/components/filelistcomp.vue [hunk 12]
```diff
@@ -0,0 +1,181 @@
...
+    },
+    data() {
+      return {
+        data:[],
+        fileslist:[]
+      }
+    },
+    methods: {
+        loadData(){
+          // this.fileslist.splice(0);
```
Comment (chrismattmann) [inline webapps/proteus-new/src/main/webapp/resources/src/components/filelistcomp.vue]: commented code. Remove.

File: webapps/proteus-new/src/main/webapp/resources/src/components/filelistcomp.vue [hunk 13]
```diff
@@ -0,0 +1,181 @@
...
+            //       });
+            
+            if(this.currentRepo!=''){
+                axios.get(this.origin+"/proteus/service/products?topn=10")
+                .then(response=>{
+                  // this.fileslist.splice(0,this.fileslist.length);
+                  
+                  response.data.forEach((v, i) => {
+                       const val = (typeof v === 'object') ? Object.assign({}, v) : v;
+                      //  this.fileslist.push(val);
```
Comment (chrismattmann) [inline webapps/proteus-new/src/main/webapp/resources/src/components/filelistcomp.vue]: commented code. Remove.

File: webapps/proteus-new/src/main/webapp/resources/src/components/filelistcomp.vue [hunk 14]
```diff
@@ -0,0 +1,181 @@
...
+                .catch(error=>{
+                  this.$log.info(error.toString()+"Progress");
+                })
+        
+            }
+            
+        }
+    },
+    computed: {
+      // fileslist(){
```
Comment (chrismattmann) [inline webapps/proteus-new/src/main/webapp/resources/src/components/filelistcomp.vue]: commented code. Remove.

File: webapps/proteus-new/src/main/webapp/resources/src/components/piechart.vue [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+
+        var path = d3.arc()
+            .outerRadius(radius - 10)
+            .innerRadius(0);
+
+        var label = d3.arc()
+            .outerRadius(radius - 40)
+            .innerRadius(radius - 40);
+
+        // d3.csv("data.csv", function(d) {
```
Comment (chrismattmann) [inline webapps/proteus-new/src/main/webapp/resources/src/components/piechart.vue]: commented code. Remove.

File: webapps/proteus-new/src/main/webapp/resources/src/components/piechart.vue [hunk 16]
```diff
@@ -0,0 +1,117 @@
...
+
+          arc.append("path")
+              .attr("d", path)
+              .attr("style", function(d) { return "fill:"+color(d.data.type) });
+
+          arc.append("text")
+              .attr("transform", function(d) { return "translate(" + label.centroid(d) + ")"; })
+              .attr("dy", "0.35em")
+              .text(function(d) { return d.data.type; });
+        // });
```
Comment (chrismattmann) [inline webapps/proteus-new/src/main/webapp/resources/src/components/piechart.vue]: commented code. Remove.

File: webapps/proteus-new/src/main/webapp/resources/src/components/projectstable.vue [hunk 17]
```diff
@@ -0,0 +1,118 @@
...
+        { text: 'Audit',sortable: false, value: 'audit' }
+      ],
+      count:{
+          numFound :0,
+          start:0
+        },
+
+        docs:[{"description":"Default DRAT repo.",
+        "text":["Default DRAT repo.","DRAT"],
+        "repo":"/home/xelvias/drat/src",
```
Comment (chrismattmann) [inline webapps/proteus-new/src/main/webapp/resources/src/components/projectstable.vue]: why not just read this JSON from the provided JSON file in the repo?

**Review by chrismattmann — COMMENTED** [review]
- see my comments. This is MUCH better. Too much commented and minified code still. Please update. Also see my comment about repo.json and why not load the initial one from the filesystem?

File: webapps/proteus-new/pom.xml (hunk 1 above)
Comment (ahmedifhaam) [reply webapps/proteus-new/pom.xml:1]: initial one ? didnt get which one ?

Comment (buggtb) [reply webapps/proteus-new/pom.xml:1]: https://github.com/apache/oodt/blob/master/pom.xml#L2-L17 like that

**Comment by buggtb** [comment]
I'm trying to process the PR but its a bit hairy. I would say, you still seem to have a minified index.html and some javascript stuff in there, shouldn't they be .gitignored and build at build time?

Also whats the licensing on those fonts you've added? I'm not sure we can check them in because of their binary nature. I'm sure @chrismattmann can tell me if I'm just making stuff up.

**Comment by chrismattmann** [comment]
@ahmedifhaam really need you to address these issues and put out a new PR in the next few days for review. Thanks.

**Comment by ahmedifhaam** [comment]
minified files in the sense ; the files inside webapp/js ?

**Comment by buggtb** [comment]
@ahmedifhaam stuff like webapp/js/app.68e075fb.js 

Isn't that a file thats been compiled during the npm build phase?

**Comment by ahmedifhaam** [comment]
yes exactly otherwise anybody wish to run have to compile it again thats why i have added that source of those files are in the resources folder should i remove the built files and build with maven ?

**Comment by buggtb** [comment]
Well I tend to believe that these repos should only contain sourcecode and if people want to build it they can do. Be it with Maven or whatever.

**Comment by ahmedifhaam** [comment]
let me see the possible alternatives

**Comment by chrismattmann** [comment]
please just make it a Maven build step to call npm. Someone shouldn't have to do anything else other than calling mvn install on the module and then the result is to call npm. This is easily done with Maven Ant Tasks, e.g., see https://gist.github.com/phillipgreenii/7c954e3c3911e5c32bd0

**Comment by ahmedifhaam** [comment]
will update soon
Thank you @chrismattmann  @buggtb

File: webapps/proteus-new/pom.xml (hunk 1 above)
Comment (ahmedifhaam) [reply webapps/proteus-new/pom.xml:1]: will add the ALv2 to all the files in next commit

File: webapps/proteus-new/src/main/webapp/resources/src/components/testii.vue [hunk 18]
```diff
@@ -0,0 +1,190 @@
+<template lang="html">
```
Comment (chrismattmann) [inline webapps/proteus-new/src/main/webapp/resources/src/components/testii.vue]: is this a test file? if so, remove it.

File: webapps/proteus-new/src/main/webapp/resources/src/components/test.vue [hunk 19]
```diff
@@ -0,0 +1,61 @@
+<template lang="html">
```
Comment (chrismattmann) [inline webapps/proteus-new/src/main/webapp/resources/src/components/test.vue]: test file? If so, remove it.

**Review by chrismattmann — COMMENTED** [review]
Close. Just needs ALv2 headers, and please remove the 2 test files. Then I think it can be committed.

File: webapps/proteus-new/src/main/webapp/resources/src/components/testii.vue (hunk 18 above)
Comment (ahmedifhaam) [reply webapps/proteus-new/src/main/webapp/resources/src/components/testii.vue]: yes already removed all the test files with next commit

**Comment by ahmedifhaam** [comment]
I have added the features promised but those need some more modifications with the next set of feedbacks i'll update.

**Comment by chrismattmann** [comment]
## Instruction for building DRAT proteus UI (new)

## Building vue front end
 1. move to `$DRAT_ROOT/webapps/proteus-new/src/main/webapp/resources`
 2. execute `npm install` ( Only on first Attempt)
 3. execute `npm run clean` 
 4. execute `npm run build`

## Building DRAT
 1. `mvn clean install`
 2. Run Drat As usual
 3. Find new UI in `DRAT_HOST/proteus-new`

After every single change to the UI step 1 and 2 has to be done.

**Comment by ahmedifhaam** [comment]
@chrismattmann we updated the build process now on _mvn clean build_ would do the build

**Comment by chrismattmann** [comment]
i was able to fully test this. For a start, it works. Which is great! Areas of improvement:

1. It does not include the DRAT reset command, or the ability to call specify commands one by one, like Proteus.
2. It does not all for git clone as Proteus does, or remote checkout of SVN or GIT
3. there is a bug in the logic for checking if all mappers have completed. I'll fix this separately.
4. the gravatar makes no sense in the upper left.
5. the views give a weird message when clicking between the audit and statistics view.

Overall great start, but a lot to work on!

**Comment by ahmedifhaam** [comment]
@chrismattmann thank you

**Pull request merged**